presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PushdownSubfields.java

```diff
@@ -284,6 +285,13 @@ public PlanNode visitOutput(OutputNode node, RewriteContext<Context> context)
             return context.defaultRewrite(node, context.get());
         }
 
+        @Override
+        public PlanNode visitCteProducer(CteProducerNode node, RewriteContext<Context> context)
```

Comment:
CteProducer functions as a data source, yet it requires the context of the output node to be inputted. I tried with a new visitor to see if it would work, but most of the tests failed